framework/set/resources/rancher2/setProvidersAndUsersTF.go

```diff
@@ -112,15 +112,15 @@ func createRequiredProviders(rootBody *hclwrite.Body, configMap []map[string]any
 
 // createProvider creates a provider block for the given rancher config.
 func createProvider(rootBody *hclwrite.Body, configMap []map[string]any, customModule bool) {
-	_, _, awsProviderVersion, linodeProviderVersion, _, _ := getRequiredProviderVersions(configMap)
+	_, _, cloudProviderVersion, linodeProviderVersion, _, _ := getRequiredProviderVersions(configMap)
 
 	terraformConfig := new(config.TerraformConfig)
 	operations.LoadObjectFromMap(config.TerraformConfigurationFileKey, configMap[0], terraformConfig)
 
 	rancherConfig := new(rancher.Config)
 	operations.LoadObjectFromMap(configs.Rancher, configMap[0], rancherConfig)
 
-	if awsProviderVersion != "" && terraformConfig.Provider == defaults.Aws && customModule {
+	if cloudProviderVersion != "" && terraformConfig.Provider == defaults.Aws && customModule {
```

Comment:
Similar to a previous comment, you will need to ensure that you're doing the exact same thing, but with `linodeProviderVersion`.